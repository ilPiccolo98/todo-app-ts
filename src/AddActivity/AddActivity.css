.float-left {
  float: left;
}

.float-right {
  float: right;
}

.status {
  clear: both;
  padding-top: 10px;
}

.submit-button {
  margin-top: 30px;
}

.error-name {
  position: relative;
  left: 33%;
}

.error-description {
  position: relative;
  left: 27%;
  margin-top: 10px;
}
