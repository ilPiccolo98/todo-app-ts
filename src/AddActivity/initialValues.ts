export type AddActivityValues = {
  name: string;
  description: string;
  status: boolean;
};

const initialValue: AddActivityValues = {
  name: "",
  description: "",
  status: false,
};

export default initialValue;
