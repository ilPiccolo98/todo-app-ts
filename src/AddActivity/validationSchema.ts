import * as yup from "yup";

const validationSchema = yup.object().shape({
  name: yup.string().required("Name is required"),
  description: yup.string().required("Description is required"),
});

export default validationSchema;
