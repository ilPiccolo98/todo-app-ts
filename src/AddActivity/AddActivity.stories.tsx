import React from "react";
import { Meta, Story } from "@storybook/react";
import { AddActivity, AddActivityProps } from "./AddActivity";

const meta: Meta = {
  title: "AddActivity",
  component: AddActivity,
  argTypes: {},
};

export default meta;
const Template: Story<AddActivityProps> = (args) => <AddActivity {...args} />;

export const Default = Template.bind({});
Default.args = {};
