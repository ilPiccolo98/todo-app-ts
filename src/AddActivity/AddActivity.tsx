import React from "react";
import { useDispatch } from "react-redux";
import { Formik, Form, Field } from "formik";
import initialValue from "./initialValues";
import { AddActivityValues } from "./initialValues";
import { addActivity } from "../activities/activitiesSlice";

const AddActivity: React.FC = (): JSX.Element => {
  const dispatch = useDispatch();
  const handleSubmit = (values: AddActivityValues) => {
    dispatch(
      addActivity({
        name: values.name,
        description: values.description,
        status: values.status,
      })
    );
  };

  return (
    <Formik initialValues={initialValue} onSubmit={handleSubmit}>
      {() => (
        <Form>
          <div>
            <label data-testid="name-label-add-activity" htmlFor="name">
              Name
            </label>
            <Field
              data-testid="name-field-add-activity"
              id="name"
              name="name"
              placeholder="Name"
            />
          </div>
          <div>
            <label
              data-testid="description-label-add-activity"
              htmlFor="description"
            >
              Description
            </label>
            <Field
              data-testid="description-field-add-activity"
              id="description"
              name="description"
              placeholder="Description"
            />
          </div>
          <div>
            <label data-testid="status-label-add-activity" htmlFor="status">
              Status
            </label>
            <Field
              data-testid="status-checkbox-add-activity"
              type="checkbox"
              id="status"
              name="status"
            />
          </div>
          <div>
            <input
              data-testid="insert-button-add-activity"
              type="submit"
              id="submit"
              name="submit"
              value="Insert"
            />
          </div>
        </Form>
      )}
    </Formik>
  );
};

export default AddActivity;
