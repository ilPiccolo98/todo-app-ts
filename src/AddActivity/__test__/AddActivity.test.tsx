import React, { MouseEvent } from "react";
import {
  fireEvent,
  render,
  RenderResult,
  waitFor,
  screen,
} from "@testing-library/react";
import AddActivity from "../AddActivity";
import { Provider } from "react-redux";
import "@testing-library/jest-dom/extend-expect";
import * as actions from "../../activities/activitiesSlice";
import store from "../../activities/activitiesStore";

const renderAddActivity = () => {
  return render(
    <Provider store={store}>
      <AddActivity />
    </Provider>
  );
};

const getAddActivityComponents = (
  component: RenderResult<
    typeof import("@testing-library/dom/types/queries"),
    HTMLElement
  >
) => {
  const nameTextField = component.getByTestId("name-addActivity").childNodes[0]
    .childNodes[1] as HTMLElement;
  const descriptionTextField = component.getByTestId("description-addActivity")
    .childNodes[0].childNodes[1] as HTMLInputElement;
  const statusCheckbox = component.getByTestId("status-addActivity")
    .childNodes[0].childNodes[0] as HTMLInputElement;
  const submit = component.getByTestId(
    "submit-addActivity"
  ) as HTMLInputElement;
  return { nameTextField, descriptionTextField, statusCheckbox, submit };
};

const activitiesSelectorSpy = jest.spyOn(actions, "activitiesSelector");
const addActivitySpy = jest.spyOn(actions, "addActivity");

describe("testing AddActivity component", () => {
  beforeEach(() => {
    activitiesSelectorSpy.mockReset().mockReturnValue([]);
    addActivitySpy.mockReset();
  });

  it("should call the addActivity action once with { id: 1, name: 'activity1', description: 'activity1', status: true, } when there is no activity", async () => {
    const component = renderAddActivity();
    const {
      nameTextField,
      descriptionTextField,
      statusCheckbox,
      submit,
    } = getAddActivityComponents(component);
    await waitFor(() =>
      fireEvent.change(nameTextField, {
        target: {
          value: "activity1",
        },
      })
    );
    await waitFor(() =>
      fireEvent.change(descriptionTextField, {
        target: {
          value: "activity1",
        },
      })
    );
    await waitFor(() => fireEvent.click(statusCheckbox));
    await waitFor(() => fireEvent.click(submit));
    expect(addActivitySpy).toHaveBeenCalledWith({
      name: "activity1",
      description: "activity1",
      status: true,
    });
  });

  it("should call the addActivity action once with { id: 2, name: 'activity2', description: 'activity2', status: true, } when there is only already an activity", async () => {
    activitiesSelectorSpy.mockReset().mockReturnValue([
      {
        id: 1,
        name: "activity1",
        description: "activity1",
        status: true,
      },
    ]);
    const component = renderAddActivity();
    const {
      nameTextField,
      descriptionTextField,
      statusCheckbox,
      submit,
    } = getAddActivityComponents(component);
    await waitFor(() =>
      fireEvent.change(nameTextField, {
        target: {
          value: "activity2",
        },
      })
    );
    await waitFor(() =>
      fireEvent.change(descriptionTextField, {
        target: {
          value: "activity2",
        },
      })
    );
    await waitFor(() => fireEvent.click(statusCheckbox));
    await waitFor(() => fireEvent.click(submit));
    expect(addActivitySpy).toHaveBeenCalledWith({
      name: "activity2",
      description: "activity2",
      status: true,
    });
  });

  it("shouldn't call the action addActivity when the name field is blank", async () => {
    const component = renderAddActivity();
    const {
      descriptionTextField,
      statusCheckbox,
      submit,
    } = getAddActivityComponents(component);
    await waitFor(() =>
      fireEvent.change(descriptionTextField, {
        target: {
          value: "activity1",
        },
      })
    );
    await waitFor(() => fireEvent.click(statusCheckbox));
    await waitFor(() => fireEvent.click(submit));
    expect(addActivitySpy).not.toBeCalled();
  });

  it("shouldn't call the action addActivity when the description field is blank", async () => {
    const component = renderAddActivity();
    const { nameTextField, statusCheckbox, submit } = getAddActivityComponents(
      component
    );
    await waitFor(() =>
      fireEvent.change(nameTextField, {
        target: {
          value: "activity1",
        },
      })
    );
    await waitFor(() => fireEvent.click(statusCheckbox));
    await waitFor(() => fireEvent.click(submit));
    expect(addActivitySpy).not.toBeCalled();
  });
});
