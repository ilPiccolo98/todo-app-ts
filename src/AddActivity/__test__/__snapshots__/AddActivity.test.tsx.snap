// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing AddActivity component should call the addActivity action once with { id: 1, name: 'activity1', description: 'activity1', status: true, } when there is no activity 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form undefined"
      >
        <p>
          <b>
            Add Activity
          </b>
        </p>
        <form
          class=""
        >
          <div>
            <div
              class="float-left"
            >
              <div
                class="secondary-container"
                data-testid="name-addActivity"
              >
                <label
                  for="name"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Name
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="name-addActivity"
                    id="name"
                    name="name"
                    placeholder="Insert name"
                    type="text"
                    value="activity1"
                    variant="secondary"
                  />
                </label>
                <div
                  class="error"
                >
                  <span
                    class="message-error undefined small"
                  >
                    <b />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="float-right"
            >
              <div
                class="secondary-container"
                data-testid="description-addActivity"
              >
                <label
                  for="description"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Description
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="description-addActivity"
                    id="description"
                    name="description"
                    placeholder="Insert description"
                    type="text"
                    value="activity1"
                    variant="secondary"
                  />
                </label>
                <div
                  class="error"
                >
                  <span
                    class="message-error undefined small"
                  >
                    <b />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="status"
            >
              <span
                class="secondary-checkbox"
                data-testid="status-addActivity"
              >
                <label
                  for="status"
                >
                  <input
                    datatestid="status-addActivity"
                    id="status"
                    label="Status"
                    name="status"
                    type="checkbox"
                    value="true"
                    variant="secondary"
                  />
                  Status
                </label>
              </span>
            </div>
            <input
              class="submit primary-submit submit-button"
              data-testid="submit-addActivity"
              datatestid="submit-addActivity"
              id="submit"
              type="submit"
              value="Add"
            />
            <div
              class="error-name"
            >
              <span
                class="message-error undefined medium"
              >
                <b />
              </span>
            </div>
            <div
              class="error-description"
            >
              <span
                class="message-error undefined medium"
              >
                <b />
              </span>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form undefined"
    >
      <p>
        <b>
          Add Activity
        </b>
      </p>
      <form
        class=""
      >
        <div>
          <div
            class="float-left"
          >
            <div
              class="secondary-container"
              data-testid="name-addActivity"
            >
              <label
                for="name"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Name
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="name-addActivity"
                  id="name"
                  name="name"
                  placeholder="Insert name"
                  type="text"
                  value="activity1"
                  variant="secondary"
                />
              </label>
              <div
                class="error"
              >
                <span
                  class="message-error undefined small"
                >
                  <b />
                </span>
              </div>
            </div>
          </div>
          <div
            class="float-right"
          >
            <div
              class="secondary-container"
              data-testid="description-addActivity"
            >
              <label
                for="description"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Description
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="description-addActivity"
                  id="description"
                  name="description"
                  placeholder="Insert description"
                  type="text"
                  value="activity1"
                  variant="secondary"
                />
              </label>
              <div
                class="error"
              >
                <span
                  class="message-error undefined small"
                >
                  <b />
                </span>
              </div>
            </div>
          </div>
          <div
            class="status"
          >
            <span
              class="secondary-checkbox"
              data-testid="status-addActivity"
            >
              <label
                for="status"
              >
                <input
                  datatestid="status-addActivity"
                  id="status"
                  label="Status"
                  name="status"
                  type="checkbox"
                  value="true"
                  variant="secondary"
                />
                Status
              </label>
            </span>
          </div>
          <input
            class="submit primary-submit submit-button"
            data-testid="submit-addActivity"
            datatestid="submit-addActivity"
            id="submit"
            type="submit"
            value="Add"
          />
          <div
            class="error-name"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
          <div
            class="error-description"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing AddActivity component should call the addActivity action once with { id: 2, name: 'activity2', description: 'activity2', status: true, } when there is only already an activity 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form undefined"
      >
        <p>
          <b>
            Add Activity
          </b>
        </p>
        <form
          class=""
        >
          <div>
            <div
              class="float-left"
            >
              <div
                class="secondary-container"
                data-testid="name-addActivity"
              >
                <label
                  for="name"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Name
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="name-addActivity"
                    id="name"
                    name="name"
                    placeholder="Insert name"
                    type="text"
                    value="activity2"
                    variant="secondary"
                  />
                </label>
                <div
                  class="error"
                >
                  <span
                    class="message-error undefined small"
                  >
                    <b />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="float-right"
            >
              <div
                class="secondary-container"
                data-testid="description-addActivity"
              >
                <label
                  for="description"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Description
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="description-addActivity"
                    id="description"
                    name="description"
                    placeholder="Insert description"
                    type="text"
                    value="activity2"
                    variant="secondary"
                  />
                </label>
                <div
                  class="error"
                >
                  <span
                    class="message-error undefined small"
                  >
                    <b />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="status"
            >
              <span
                class="secondary-checkbox"
                data-testid="status-addActivity"
              >
                <label
                  for="status"
                >
                  <input
                    datatestid="status-addActivity"
                    id="status"
                    label="Status"
                    name="status"
                    type="checkbox"
                    value="true"
                    variant="secondary"
                  />
                  Status
                </label>
              </span>
            </div>
            <input
              class="submit primary-submit submit-button"
              data-testid="submit-addActivity"
              datatestid="submit-addActivity"
              id="submit"
              type="submit"
              value="Add"
            />
            <div
              class="error-name"
            >
              <span
                class="message-error undefined medium"
              >
                <b />
              </span>
            </div>
            <div
              class="error-description"
            >
              <span
                class="message-error undefined medium"
              >
                <b />
              </span>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form undefined"
    >
      <p>
        <b>
          Add Activity
        </b>
      </p>
      <form
        class=""
      >
        <div>
          <div
            class="float-left"
          >
            <div
              class="secondary-container"
              data-testid="name-addActivity"
            >
              <label
                for="name"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Name
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="name-addActivity"
                  id="name"
                  name="name"
                  placeholder="Insert name"
                  type="text"
                  value="activity2"
                  variant="secondary"
                />
              </label>
              <div
                class="error"
              >
                <span
                  class="message-error undefined small"
                >
                  <b />
                </span>
              </div>
            </div>
          </div>
          <div
            class="float-right"
          >
            <div
              class="secondary-container"
              data-testid="description-addActivity"
            >
              <label
                for="description"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Description
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="description-addActivity"
                  id="description"
                  name="description"
                  placeholder="Insert description"
                  type="text"
                  value="activity2"
                  variant="secondary"
                />
              </label>
              <div
                class="error"
              >
                <span
                  class="message-error undefined small"
                >
                  <b />
                </span>
              </div>
            </div>
          </div>
          <div
            class="status"
          >
            <span
              class="secondary-checkbox"
              data-testid="status-addActivity"
            >
              <label
                for="status"
              >
                <input
                  datatestid="status-addActivity"
                  id="status"
                  label="Status"
                  name="status"
                  type="checkbox"
                  value="true"
                  variant="secondary"
                />
                Status
              </label>
            </span>
          </div>
          <input
            class="submit primary-submit submit-button"
            data-testid="submit-addActivity"
            datatestid="submit-addActivity"
            id="submit"
            type="submit"
            value="Add"
          />
          <div
            class="error-name"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
          <div
            class="error-description"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing AddActivity component shouldn't call the action addActivity when the description field is blank 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form undefined"
      >
        <p>
          <b>
            Add Activity
          </b>
        </p>
        <form
          class=""
        >
          <div>
            <div
              class="float-left"
            >
              <div
                class="secondary-container"
                data-testid="name-addActivity"
              >
                <label
                  for="name"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Name
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="name-addActivity"
                    id="name"
                    name="name"
                    placeholder="Insert name"
                    type="text"
                    value="activity1"
                    variant="secondary"
                  />
                </label>
                <div
                  class="error"
                >
                  <span
                    class="message-error undefined small"
                  >
                    <b />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="float-right"
            >
              <div
                class="secondary-container"
                data-testid="description-addActivity"
              >
                <label
                  for="description"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Description
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="description-addActivity"
                    id="description"
                    name="description"
                    placeholder="Insert description"
                    type="text"
                    value=""
                    variant="secondary"
                  />
                </label>
                <div
                  class="error"
                >
                  <span
                    class="message-error undefined small"
                  >
                    <b />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="status"
            >
              <span
                class="secondary-checkbox"
                data-testid="status-addActivity"
              >
                <label
                  for="status"
                >
                  <input
                    datatestid="status-addActivity"
                    id="status"
                    label="Status"
                    name="status"
                    type="checkbox"
                    value="true"
                    variant="secondary"
                  />
                  Status
                </label>
              </span>
            </div>
            <input
              class="submit primary-submit submit-button"
              data-testid="submit-addActivity"
              datatestid="submit-addActivity"
              id="submit"
              type="submit"
              value="Add"
            />
            <div
              class="error-name"
            >
              <span
                class="message-error undefined medium"
              >
                <b />
              </span>
            </div>
            <div
              class="error-description"
            >
              <span
                class="message-error undefined medium"
              >
                <b>
                  Description is required
                </b>
              </span>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form undefined"
    >
      <p>
        <b>
          Add Activity
        </b>
      </p>
      <form
        class=""
      >
        <div>
          <div
            class="float-left"
          >
            <div
              class="secondary-container"
              data-testid="name-addActivity"
            >
              <label
                for="name"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Name
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="name-addActivity"
                  id="name"
                  name="name"
                  placeholder="Insert name"
                  type="text"
                  value="activity1"
                  variant="secondary"
                />
              </label>
              <div
                class="error"
              >
                <span
                  class="message-error undefined small"
                >
                  <b />
                </span>
              </div>
            </div>
          </div>
          <div
            class="float-right"
          >
            <div
              class="secondary-container"
              data-testid="description-addActivity"
            >
              <label
                for="description"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Description
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="description-addActivity"
                  id="description"
                  name="description"
                  placeholder="Insert description"
                  type="text"
                  value=""
                  variant="secondary"
                />
              </label>
              <div
                class="error"
              >
                <span
                  class="message-error undefined small"
                >
                  <b />
                </span>
              </div>
            </div>
          </div>
          <div
            class="status"
          >
            <span
              class="secondary-checkbox"
              data-testid="status-addActivity"
            >
              <label
                for="status"
              >
                <input
                  datatestid="status-addActivity"
                  id="status"
                  label="Status"
                  name="status"
                  type="checkbox"
                  value="true"
                  variant="secondary"
                />
                Status
              </label>
            </span>
          </div>
          <input
            class="submit primary-submit submit-button"
            data-testid="submit-addActivity"
            datatestid="submit-addActivity"
            id="submit"
            type="submit"
            value="Add"
          />
          <div
            class="error-name"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
          <div
            class="error-description"
          >
            <span
              class="message-error undefined medium"
            >
              <b>
                Description is required
              </b>
            </span>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing AddActivity component shouldn't call the action addActivity when the name field is blank 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form undefined"
      >
        <p>
          <b>
            Add Activity
          </b>
        </p>
        <form
          class=""
        >
          <div>
            <div
              class="float-left"
            >
              <div
                class="secondary-container"
                data-testid="name-addActivity"
              >
                <label
                  for="name"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Name
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="name-addActivity"
                    id="name"
                    name="name"
                    placeholder="Insert name"
                    type="text"
                    value=""
                    variant="secondary"
                  />
                </label>
                <div
                  class="error"
                >
                  <span
                    class="message-error undefined small"
                  >
                    <b />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="float-right"
            >
              <div
                class="secondary-container"
                data-testid="description-addActivity"
              >
                <label
                  for="description"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Description
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="description-addActivity"
                    id="description"
                    name="description"
                    placeholder="Insert description"
                    type="text"
                    value="activity1"
                    variant="secondary"
                  />
                </label>
                <div
                  class="error"
                >
                  <span
                    class="message-error undefined small"
                  >
                    <b />
                  </span>
                </div>
              </div>
            </div>
            <div
              class="status"
            >
              <span
                class="secondary-checkbox"
                data-testid="status-addActivity"
              >
                <label
                  for="status"
                >
                  <input
                    datatestid="status-addActivity"
                    id="status"
                    label="Status"
                    name="status"
                    type="checkbox"
                    value="true"
                    variant="secondary"
                  />
                  Status
                </label>
              </span>
            </div>
            <input
              class="submit primary-submit submit-button"
              data-testid="submit-addActivity"
              datatestid="submit-addActivity"
              id="submit"
              type="submit"
              value="Add"
            />
            <div
              class="error-name"
            >
              <span
                class="message-error undefined medium"
              >
                <b>
                  Name is required
                </b>
              </span>
            </div>
            <div
              class="error-description"
            >
              <span
                class="message-error undefined medium"
              >
                <b />
              </span>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form undefined"
    >
      <p>
        <b>
          Add Activity
        </b>
      </p>
      <form
        class=""
      >
        <div>
          <div
            class="float-left"
          >
            <div
              class="secondary-container"
              data-testid="name-addActivity"
            >
              <label
                for="name"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Name
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="name-addActivity"
                  id="name"
                  name="name"
                  placeholder="Insert name"
                  type="text"
                  value=""
                  variant="secondary"
                />
              </label>
              <div
                class="error"
              >
                <span
                  class="message-error undefined small"
                >
                  <b />
                </span>
              </div>
            </div>
          </div>
          <div
            class="float-right"
          >
            <div
              class="secondary-container"
              data-testid="description-addActivity"
            >
              <label
                for="description"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Description
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="description-addActivity"
                  id="description"
                  name="description"
                  placeholder="Insert description"
                  type="text"
                  value="activity1"
                  variant="secondary"
                />
              </label>
              <div
                class="error"
              >
                <span
                  class="message-error undefined small"
                >
                  <b />
                </span>
              </div>
            </div>
          </div>
          <div
            class="status"
          >
            <span
              class="secondary-checkbox"
              data-testid="status-addActivity"
            >
              <label
                for="status"
              >
                <input
                  datatestid="status-addActivity"
                  id="status"
                  label="Status"
                  name="status"
                  type="checkbox"
                  value="true"
                  variant="secondary"
                />
                Status
              </label>
            </span>
          </div>
          <input
            class="submit primary-submit submit-button"
            data-testid="submit-addActivity"
            datatestid="submit-addActivity"
            id="submit"
            type="submit"
            value="Add"
          />
          <div
            class="error-name"
          >
            <span
              class="message-error undefined medium"
            >
              <b>
                Name is required
              </b>
            </span>
          </div>
          <div
            class="error-description"
          >
            <span
              class="message-error undefined medium"
            >
              <b />
            </span>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
